sdk/eventhub/Azure.Messaging.EventHubs/src/Core/TransportProducerPool.cs

```diff
@@ -0,0 +1,396 @@
+                            }
+                        }
+                    }
+                }
+            };
+        }
+
+        /// <summary>
+        ///   An item of the pool, adding tracking information to a <see cref="TransportProducer" />.
+        /// </summary>
+        ///
+        internal class PoolItem
+        {
+            /// <summary>The period of inactivity after which an item would be removed from the pool.</summary>
+            internal static readonly TimeSpan DefaultRemoveAfterDuration = TimeSpan.FromMinutes(10);
+
+            /// <summary>
+            ///   An abstracted Event Hub transport-specific <see cref="TransportProducer" /> that is associated with a specific partition.
+            /// </summary>
+            ///
+            public TransportProducer PartitionProducer { get; private set; }
+
+            /// <summary>
+            ///   The unique identifier of a partition associated with the Event Hub.
+            /// </summary>
+            ///
+            public string PartitionId { get; private set; }
+
+            /// <summary>
+            ///   A set of unique identifiers used to track which instances of a <see cref="PoolItem" /> are active.
+            /// </summary>
+            ///
+            public ConcurrentDictionary<string, byte> ActiveInstances { get; } = new ConcurrentDictionary<string, byte>();
+
+            /// <summary>
+            ///   The UTC date and time when a <see cref="PoolItem" /> will become eligible for eviction.
+            /// </summary>
+            ///
+            public DateTimeOffset RemoveAfter { get; set; }
+
+            /// <summary>
+            ///   Extends the UTC date and time when <see cref="PoolItem" /> will become eligible for eviction.
+            /// </summary>
+            ///
+            /// <param name="removeAfterDuration">The period of inactivity after which a <see cref="PoolItem" /> will become eligible for eviction.</param>
+            ///
+            public void ExtendRemoveAfter(TimeSpan? removeAfterDuration)
+            {
+                RemoveAfter = DateTimeOffset.UtcNow.Add(removeAfterDuration ?? DefaultRemoveAfterDuration);
+            }
+
+            /// <summary>
+            ///   Creates a <see cref="TransportProducer" /> matching the <see cref="PartitionId" />.
+            /// </summary>
+            ///
+            /// <param name="connection">The <see cref="EventHubConnection" /> connection to use for communication with the Event Hubs service.</param>
+            /// <param name="retryPolicy">The policy to use for determining retry behavior for when an operation fails.</param>
+            ///
+            /// <returns>A <see cref="TransportProducer" /> matching the <see cref="PartitionId" />.</returns>
+            ///
+            public TransportProducer GetTransportProducer(EventHubConnection connection,
```

Comment:
If the pool owns the connection and retry policy, can we just inject a transport producer when the pool item is created, or pass a factory into the pool item?    I'm trying to remember why we have this as `public`.   Looking now, I would expect a caller to be able to just hit the property and have this run automatically..... but I may just not be remembering something.

Reply:
I removed this method, it was instrumental to allow the a ```PoolItem``` to retrieve the right ```TransportProducer``` and also to retrieve the ```EventHubTransportProducer``` when a batch is created in ```EventHubProducerClient.CreateBatchAsync```